src/D7/FileDownloadTrait.php

```diff
@@ -318,4 +322,26 @@ protected function fileDownloadGetTempDir() {
     return '/tmp/behat_downloads';
   }
 
+  /**
+   * Before nodeCreate set field value for file.
+   *
+   * Field should be in format 'file;__file_source__;__file_name_.
+   *
+   * @beforeNodeCreate
+   */
+  public function nodeCreateAlter(EntityScope $scope) {
+    $node = $scope->getEntity();
+    foreach ($node as $key => $value) {
+      if (strpos($value, 'file;') !== FALSE) {
+        $file_info = explode(';', $value);
+        $file_source = $file_info[1];
+        $file_name = $file_info[2];
+        $image = file_get_contents($file_source);
+        file_save_data($image, 'public://' . $file_name, FILE_EXISTS_RENAME);
+
+        $node->{$key} = $file_name;
+      }
+    }
+  }
+
```

Comment:
It test was firstly by priority, but the next(FileTrait) has "given files:" defenition. Should I remove this code and relation for it(Changes in MakeFile) and replace it with "given files:"? Cause I didn't know about this defenition before I started  next test

Reply:
yes.

`given files` actually creates files objects, but you still need to attach those objects to fields - see example below.